File: 0062-unique-paths/0062-unique-paths.cpp
class Solution {
public:
    int uniquePaths(int m, int n) {
        vector<vector<int>> dp(m, vector<int>(n,1));
        for (int row = 1; row < m; row++) {
            for (int col = 1; col < n; col++) {
                dp[row][col] = dp[row-1][col] + dp[row][col-1];
            }
        }
        return dp[m-1][n-1];
    }
};
//class Solution {
//public:
//    int uniquePaths(int m, int n) {
//        if(m == 1 || n == 1) return 1;
//        return uniquePaths(m-1, n) + uniquePaths(m, n-1);
//    }
//};

File: 0062-unique-paths/0062-unique-paths.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        if m == 1 or n == 1:
            return 1
        dp = [[0] * n for _ in range(m)]
        for row in range(m):
            dp[row][0] = 1
        for col in range(n):
            dp[0][col] = 1
        for row in range(1, m):
            for col in range(1, n):
                dp[row][col] = dp[row][col-1] + dp[row-1][col]
        return dp[-1][-1]
